s4e-web/src/app/views/map-view/view-manager/view-manager.component.ts

```diff
@@ -1,23 +1,56 @@
-import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
 import {IUILayer} from '../state/common.model';
+import {FormControl} from '@ng-stack/forms';
+import {SearchResult} from '../state/search-results/search-result.model';
+import {SearchResultsQuery} from '../state/search-results/search-results.query';
+import {environment} from '../../../../environments/environment';
+import {debounceTime, distinctUntilChanged} from 'rxjs/operators';
 
 @Component({
   selector: 's4e-view-manager',
   templateUrl: './view-manager.component.html',
   styleUrls: ['./view-manager.component.scss'],
 })
-export class ViewManagerComponent {
+export class ViewManagerComponent implements OnInit{
   @Input() loading = true;
   @Input() userLoggedIn: boolean = false;
   @Input() products: IUILayer[] = [];
   @Input() productsTypes: IUILayer[] = [];
   @Input() productTypeLoading: boolean = true;
   @Output() selectProductType = new EventEmitter<number>();
+  @Output() changedSearchFocus = new EventEmitter<boolean>();
+
   @Input() overlays: IUILayer[] = [];
   @Input() overlaysLoading: boolean = true;
-
   @Output() selectOverlay = new EventEmitter<string>();
   @Output() logout = new EventEmitter<void>();
 
-  constructor() {}
+  @Output() searchForPlaces = new EventEmitter<string>();
+  @Input() searchResults: SearchResult[];
+  @Input() searchResultsLoading: boolean;
+  @Input() searchResultsOpen: boolean;
+  @Output() searchResultClicked = new EventEmitter<SearchResult>();
+  searchFc: FormControl<string> = new FormControl<string>('');
+
+  constructor(private searchResultQuery: SearchResultsQuery) {
+  }
+
+  ngOnInit(): void {
+    if (environment.hmr) {
+      this.searchFc.setValue(this.searchResultQuery.getValue().queryString);
+    }
+
+    this.searchFc.valueChanges.pipe(
+      debounceTime(300),
+      distinctUntilChanged()
+    ).subscribe((text: string) => this.searchForPlaces.emit(text));
```

Comment:
Subscriptions must be unsubscribed in `ngOnDestroy`, it can be easily done (without storing subscription reference using a helper `untilDestroy`:

```
    this.searchFc.valueChanges.pipe(
        debounceTime(300),
        distinctUntilChanged(),
        untilDestroyed(this),
      ).subscribe((text: string) => this.searchForPlaces.emit(text));
```

check out usage in other components, that should provide you with necessary information